packages/@aws-cdk/aws-codedeploy/lib/lambda/application.ts

```diff
@@ -0,0 +1,100 @@
+import cdk = require("@aws-cdk/cdk");
+import { CfnApplication } from "../codedeploy.generated";
+import { ComputePlatform } from "../config";
+import { applicationNameToArn } from "../utils";
+
+/**
+ * Represents a reference to a CodeDeploy Application deploying to AWS Lambda.
+ *
+ * If you're managing the Application alongside the rest of your CDK resources,
+ * use the {@link LambdaApplication} class.
+ *
+ * If you want to reference an already existing Application,
+ * or one defined in a different CDK Stack,
+ * use the {@link #import} method.
+ */
+export interface ILambdaApplication {
+  readonly applicationArn: string;
+  readonly applicationName: string;
+  readonly computePlatform: ComputePlatform.Lambda;
```

Comment:
I don't understand why is this property here...? A Lambda Application obviously has the Lambda Compute Platform. In fact, we never surfaced anything like 'Compute Platform' with Server Applications, and I don't believe we should surface it now.

Reply:
So this is a subtle problem with structural (TypeScript) vs nominal (Java, C#, etc.) type systems. In TypeScript, the following types are identical, as they are distinguished by their shape, not their name. In Java/C# they are different types.

```ts
export interface ILambdaApplication {
  readonly applicationArn: string;
  readonly applicationName: string;
  export(): LambdaApplicationImportProps;
}
```

```ts
export interface IServerApplication extends cdk.IConstruct {
  readonly applicationArn: string;
  readonly applicationName: string;
  export(): ServerApplicationImportProps;
}
```

This means that you can do the following without any type-checking. It will break in a nominal type system but not in a structural one.

```ts
const serverApp: IServerApplication = ...;
const lambdaApp: ILambdaApplication = LambdaApplication.import(serverApp.export());

// or
const lambdaApp: ILambdaApplication = serverApp;
```

That's why i added compute platform as a type-tag. Notice how I'm using the literal type, `ComputePlatform.Server` and `ComputePlatform.Lambda`, not just `ComputePlatform`:

```ts
export interface ILambdaApplication {
  readonly applicationArn: string;
  readonly applicationName: string;
  readonly computePlatform: ComputePlatform.Lambda; // <- Literal type

  export(): LambdaApplicationImportProps;
}
```

This literal type is erased by JSII, but that doesn't matter because the languages we export to are either dynamic or use nominal type systems. This is the only way to achieve consistent type-safety for all supported languages.

That said, we should discuss whether we care, especially if it's polluting the API.

Reply:
I don't think it's worth it. In TypeScript, you're one `as any` away from doing whatever you want anyway.

I can see this sort of marker done as a private member, but definitely not as part of the public API of the Construct.